connexion/middleware/security.py

```diff
@@ -0,0 +1,229 @@
+import logging
+import pathlib
+import typing as t
+from collections import defaultdict
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+from connexion.apis.abstract import AbstractSpecAPI
+from connexion.exceptions import MissingMiddleware
+from connexion.http_facts import METHODS
+from connexion.lifecycle import MiddlewareRequest
+from connexion.middleware import AppMiddleware
+from connexion.security import SecurityHandlerFactory
+
+logger = logging.getLogger("connexion.middleware.security")
+
+
+class SecurityMiddleware(AppMiddleware):
+    """Middleware to check if operation is accessible on scope."""
+
+    def __init__(self, app: ASGIApp) -> None:
+        self.app = app
+        self.apis: t.Dict[str, SecurityAPI] = {}
+
+    def add_api(self, specification: t.Union[pathlib.Path, str, dict], **kwargs) -> None:
+        api = SecurityAPI(specification, **kwargs)
+        self.apis[api.base_path] = api
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send):
+        if scope["type"] != "http":
+            await self.app(scope, receive, send)
+            return
+
+        try:
+            connexion_context = scope['extensions']['connexion_routing']
```

Comment:
Would like to see this extension/context name defined in the RoutingMiddleware as I see this is directly  linked with that. For example RoutingMiddleware.CONTEXT_NAME. Means less prone to typos and auto complete.

Reply:
Good point, lost this while refactoring.